diagnostics/physics/plot_common.py

```diff
@@ -225,3 +233,98 @@ def process_glorys(config, target_grid):
 
     logger.info("Glorys data processed successfully.")
     return glorys_rg, glorys_ave, glorys_lonc, glorys_latc
+
+def get_end_of_climatology_period(clima_file):
+    """
+    Determine the time period covered by the last climatology file. This function is needed
+    because decadal climatology data may not be a decade long for the last available time
+    period - for example, the last decadal climatology file for the NWA domain extends from
+    2005 to 2017. If the climatology files are formatted in the standard NCEI format, the
+    last year of the decadal file should be readable from the last two characters of the
+    second _ separated entry in the file name.
+
+    """
+    # Extract the time period information from the given file path
+    logger.info(f"Finding the end year of the last climatology file using file name: {clima_file}")
+    period = os.path.basename(clima_file).split("_")[1]
+    logger.info(f"Climatology time period: {period}")
+    end_decade_char = period[-2]
+    logger.info(f"Climatology decade character: {end_decade_char}")
+    end_year_char = period[-1]
+    logger.info(f"Climatology year character: {end_year_char}")
+
+    # Map the derived decade and year characters to an actual year, stored as an int
+    if end_decade_char == "A":
+        end_year = 2000 + int(end_year_char)
+    elif end_decade_char == "B":
+        end_year = 2010 + int(end_year_char)
+    elif end_decade_char == "C":
+        end_year = 2020 + int(end_year_char)
+    else:
+        logger.error((f"ERROR: Found unrecognized decade code {end_decade_char} in {clima_file}."
+        "please make sure that decadal climatology follows the National Centers for Environemntal"
+        "Information naming conventions"))
+        raise Exception("Error: unrecognized decade code {end_decade_char} in {clima_file}")
```

Comment:
Perhaps we can refactor this part as:
```
    # Map the derived decade and year characters to an actual year, stored as an int
    decade_mapping = {
        "A": 2000,
        "B": 2010,
        "C": 2020
    }

    if end_decade_char in decade_mapping:
        end_year = decade_mapping[end_decade_char] + int(end_year_char)
    else:
        logger.error((f"ERROR: Found unrecognized decade code {end_decade_char} in {clima_file}."
        "please make sure that decadal climatology follows the National Centers for Environemntal"
        "Information naming conventions"))
        raise ValueError(f"Error: Unrecognized decade code {end_decade_char} in {clima_file}")
```
This will make it easier to maintain in the future.